docs/faq.md

```diff
@@ -0,0 +1,98 @@
+# Frequently Asked Questions
+
+## How do I customize X on the worker group's settings?
+
+All the options that can be customized for worker groups are listed in [local.tf](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/local.tf) under `workers_group_defaults_defaults`.
+
+Please open Issues or PRs if you think something is missing.
+
+## Why are nodes not being registered?
+
+### Networking
+
+Often caused by a networking or endpoint configuration issue.
+
+At least one of the cluster public or private endpoints must be enabled in order for access to the cluster to work.
+
+Your nodes need to be able to contact the EKS cluster endpoint. By default the module only creates a public endpoint. You should also enable the private endpoint by setting `cluster_endpoint_private_access = true` on this module.
+
+If you have the private endpoint enabled ensure that you also have VPC DNS enabled. Set `enable_dns_hostnames = true` on your `aws_vpc` resource or the `terraform-aws-module/vpc/aws` community module.
+
+Nodes need to be able to connect to AWS services plus pull down container images from repos. You can either:
+- enable endpoints to the relevant services, if only using ECR repos and for some reason cannot enable public outbound: EC2 API, ECR API, ECR DKR and S3
+- enable outbound public internet access:
+  - Private subnets: via a NAT gateway or instance
+  - Public subnets: assign public IPs to nodes
+
+### `aws-auth` ConfigMap not present
+
+The module configures the `aws-auth` ConfigMap. This is used by the cluster to grant IAM users RBAC permissions in the cluster. Sometimes the map fails to apply correctly, especially if terraform could not access the cluster endpoint during cluster creation.
+
+Confirm that the ConfigMap matches the contents of the generated `config-map-aws-auth_${cluster_name}.yaml` file. You can retrieve the live config by running the following in your terraform folder:
+`kubectl --kubeconfig=kubeconfig_* -n kube-system get cm aws-auth -o yaml`
+
+Apply the config with:
+`kubectl --kubeconfig=kubeconfig_* apply -f config-map-aws-auth_*.yaml`
+
+## How can I work with the cluster if I disable the public endpoint?
+
+You have to interact with the cluster from within the VPC that it's associated with, from an instance that's allowed access via the cluster's security group.
+
+Creating a new cluster with the public endpoint disabled is harder to achieve. You will either want to pass in a pre-configured cluster security group or apply the `aws-auth` configmap in a separate action.
+
+## How can I stop Terraform from removing the EKS tags from my VPC and subnets?
```

Comment:
@dpiddockcmp tag removal must be made compulsory form the subnets because on `aws` i faced an issue that the nodes are not being deployed because of the existing tags conflicts. I would be good to add this in FAQ too